dotcom-rendering/src/web/lib/contributions.ts

```diff
@@ -183,6 +184,21 @@ export const hasOptedOutOfArticleCount = async (): Promise<boolean> => {
 	return !hasCmpConsent || hasArticleCountOptOutCookie();
 };
 
+// A hook to find out if a user has opted out of article counting
+export const useHasOptedOutOfArticleCount = (): boolean | 'Pending' => {
+	const [hasOptedOut, setHasOptedOut] = useState<boolean | 'Pending'>(
+		'Pending',
+	);
+
+	useEffect(() => {
```

Comment:
DCR has a `useOnce` hook, which I think you could use in place of `useEffect` with an empty dependency array. I like it as the name makes it clear what the behaviour is.